packages/langium/src/workspace/workspace-lock.ts

```diff
@@ -59,8 +64,25 @@ export class DefaultWorkspaceLock implements WorkspaceLock {
         return this.enqueue(this.writeQueue, action, tokenSource.token);
     }
 
-    read<T>(action: () => MaybePromise<T>): Promise<T> {
-        return this.enqueue(this.readQueue, action);
+    read<T>(action: () => MaybePromise<T>, priority?: boolean): Promise<T> {
+        if (priority) {
+            this.counter++;
+            const deferred = new Deferred<T>();
+            const end = (run: () => void) => {
+                run();
+                // Ensure that in any case the counter is decremented and the next operation is performed
+                this.counter--;
+                this.performNextOperation();
+            };
+            // Instanly run the action and resolve/reject the promise afterwards
+            Promise.resolve(action()).then(
+                result => end(() => deferred.resolve(result)),
+                err => end(() => deferred.reject(err))
+            );
```

Comment:
I see two potential issues with this code:
- The solution with the local `end` function is a more complicated way of expressing a `finally` block.
- I suspect that `Promise.resolve(action())` does not behave as we want when the `action` throws an error – the error would just be propagated instead of rejecting the resulting promise.

I suggest making this whole method `async` so we can `await` the action and wrap that in a `try`-`finally` block.